internal/controller/controllers/bmh_agent_controller.go

```diff
@@ -72,12 +72,21 @@ const (
 type reconcileResult interface {
 	Result() (reconcile.Result, error)
 	Dirty() bool
+	Stop() bool
 }
 
 // reconcileComplete is a result indicating that the current reconcile has completed,
 // and there is nothing else to do.
+//
+// Set the `dirty` flag when the BMH CR (or any CR) has been modified and an `Update`
```

Comment:
These docs might be more useful on the interface methods, rather than here on just one implementation.